Source/Core/Core/HW/BBA-TAP/TAP_Win32.cpp

```diff
@@ -263,78 +284,69 @@ bool CEXIETHERNET::SendFrame(u8 *frame, u32 size)
 	return true;
 }
 
-VOID CALLBACK CEXIETHERNET::ReadWaitCallback(PVOID lpParameter, BOOLEAN TimerFired)
+static void ReadThreadHandler(CEXIETHERNET* self)
 {
-	CEXIETHERNET* self = (CEXIETHERNET*)lpParameter;
+	while (true)
+	{
+		if (self->mHAdapter == INVALID_HANDLE_VALUE)
+			return;
 
-	GetOverlappedResult(self->mHAdapter, &self->mReadOverlapped,
-		(LPDWORD)&self->mRecvBufferLength, false);
+		DWORD res = ReadFile(self->mHAdapter, self->mRecvBuffer, BBA_RECV_SIZE, (LPDWORD)&self->mRecvBufferLength, &self->mReadOverlapped);
```

Comment:
Is the cast to `LPDWORD` (I believe that's the equivalent of `unsigned int*`) necessary here?